engine/execution/ingestion/core.go

```diff
@@ -88,20 +89,25 @@ func NewCore(
 		collectionFetcher: collectionFetcher,
 		eventConsumer:     eventConsumer,
 	}
+
+	e.ComponentManager = component.NewComponentManagerBuilder().
+		AddWorker(e.launchWorker).
```

Comment:
I figured out. 

The reason it wasn't working, is because I thought the Component is only expecting the machine to return anything that implements the `Ready` and `Done` function for `ReadyDoneAware` interface. That's why I let `machine` to return the `Ready` and `Done` function from the `Core`. But actually, it not long depend on the `ReadyDoneAware`, it is expecting to return a Component itself, and then the component manager [will cast the `ReadyDoneAware` interface back to a Component,](https://github.com/onflow/flow-go/blob/master/cmd/scaffold.go#L1717) and then call `RunComponent` on it. 

I fixed it by letting machine returning the Core as a Component manager.